libraries/AP_Scripting/generator/src/main.c

```diff
@@ -1844,9 +1829,20 @@ void emit_singleton_fields() {
 int emit_references(const struct argument *arg, const char * tab) {
   int arg_index = NULLABLE_ARG_COUNT_BASE + 2;
   int return_count = 0;
+  // count arguments to return so we know if we need to check the stack
+  const struct argument *count_arg = arg;
+  while (count_arg != NULL) {
+    if (count_arg->type.flags & (TYPE_FLAGS_NULLABLE | TYPE_FLAGS_REFERNCE)) {
+      return_count++;
+    }
+    count_arg = count_arg->next;
+  }
+  // add one to have a spare stack slot for userdata creation funcs
+  fprintf(source, "%sif (%d > LUA_MINSTACK) {\n", tab, return_count+1);
```

Comment:
I guess this could be a `#if` too. Hopefully the compiler is smart enough that the result is the same.

Reply:
Makes more sense, done.